lib/immobilienscout/api/report.rb

```diff
@@ -30,6 +26,31 @@ def query_params(date_from, date_to)
         def retrieve_url(is24_id)
           "#{Immobilienscout::Client.api_url}/restapi/api/offer/v1.0/user/me/realestate/#{is24_id}/dailyreport"
         end
+
+        def fix(parsed_response)
+          response_array_fixed = place_report_daily_data_in_array(parsed_response)
+          convert_numers_from_string_to_integers(response_array_fixed)
+        end
+
+        def place_report_daily_data_in_array(parsed_response)
+          report_daily_data = parsed_response.messages['dailyReports']['reportDailyData']
+
+          if report_daily_data.is_a? Hash
+          parsed_response.messages['dailyReports']['reportDailyData'] = [report_daily_data]
+          end
+
+          parsed_response
+        end
+
+        def convert_numers_from_string_to_integers(parsed_response)
+          parsed_response.messages['dailyReports']['reportDailyData'].map do |hash|
+            hash.except('date').map do |key, value|
```

Comment:
For both loops, we should use `each` instead of `map`, because we're not interested in the result of the iteration. We're relying on assignment here and will return the mutated response at the end anyway.